lib/test-unit.rb

```diff
@@ -4,8 +4,21 @@
 
 module Test
   module Unit
-    autoload :TestCase, "test/unit/testcase"
-    autoload :AutoRunner, "test/unit/autorunner"
+    LAZY_LOADER_MAPPING = {
+      :TestCase => "test/unit/testcase",
+      :AutoRunner => "test/unit/autorunner",
+    }
+
+    class << self
+      def const_missing(name)
+        if LAZY_LOADER_MAPPING.key?(name)
+          require_relative LAZY_LOADER_MAPPING[name]
+          const_get(name)
+        else
+          super
+        end
+      end
+    end
```

Comment:
Can we remove this `const_missing` when it's no longer needed?

```suggestion
    class << self
      def const_missing(name)
        case name
        when :AutoRunner, :TestCase
          require_relative "test/unit/autorunner"
          require_relative "test/unit/testcase"
          singleton_class.remove_method(:const_missing)
          const_get(name)
        else
          super
        end
      end
    end
```